src/custom-elements.js

```diff
@@ -646,37 +672,42 @@ let Deferred;
   CustomElementRegistry.prototype['_observeRoot'] = CustomElementRegistry.prototype._observeRoot;
   CustomElementRegistry.prototype['_addImport'] = CustomElementRegistry.prototype._addImport;
 
-  // patch window.HTMLElement
+  patchElement('HTMLElement')
+  var htmlElementSubclasses = [ 'Button', 'Canvas', 'Data', 'Head', 'Mod', 'TableCell', 'TableCol', 'Anchor', 'Area', 'Base', 'Body', 'BR', 'DataList', 'Details', 'Dialog', 'Div', 'DList', 'Embed', 'FieldSet', 'Form', 'Heading', 'HR', 'Html', 'IFrame', 'Image', 'Input', 'Keygen', 'Label', 'Legend', 'LI', 'Link', 'Map', 'Media', 'Menu', 'MenuItem', 'Meta', 'Meter', 'Object', 'OList', 'OptGroup', 'Option', 'Output', 'Paragraph', 'Param', 'Picture', 'Pre', 'Progress', 'Quote', 'Script', 'Select', 'Slot', 'Source', 'Span', 'Style', 'TableCaption', 'Table', 'TableRow', 'TableSection', 'Template', 'TextArea', 'Time', 'Title', 'Track', 'UList', 'Unknown'];
+  for (let index in htmlElementSubclasses) {
+      patchElement(`HTML${htmlElementSubclasses[Number(index)]}Element`);
+  }
 
-  /** @const */
-  const origHTMLElement = window.HTMLElement;
-  CustomElementRegistry.prototype['nativeHTMLElement'] = origHTMLElement;
-  /**
-   * @type {function(new: HTMLElement)}
-   */
-  const newHTMLElement = function HTMLElement() {
-    const customElements = _customElements();
+  function patchElement(varName) {
+      /** @const */
+      const origHTMLElement = window[varName];
+      if (!origHTMLElement) {
+          return;
+      }
+      CustomElementRegistry.prototype[`native${varName}`] = origHTMLElement;
+      /**
+       * @type {function(new: HTMLElement)}
+       */
+      const newHTMLElement = function() {
+        const customElements = _customElements();
 
-    // If there's an being upgraded, return that
-    if (customElements._newInstance) {
-      const i = customElements._newInstance;
-      customElements._newInstance = null;
-      return i;
-    }
-    if (this.constructor) {
-      // Find the tagname of the constructor and create a new element with it
-      const tagName = customElements._constructors.get(this.constructor);
-      return _createElement(document, tagName, undefined, false);
-    }
-    throw new Error('Unknown constructor. Did you call customElements.define()?');
+        // If there's an being upgraded, return that
+        if (customElements._newInstance) {
+          const i = customElements._newInstance;
+          customElements._newInstance = null;
+          return i;
+        }
+        if (this.constructor) {
+          // Find the tagname of the constructor and create a new element with it
+          const constructorInfo = customElements._constructors.get(this.constructor);
```

Comment:
We need more info than just the `tagName`, so that we know if we're creating a customized built-in element or not. See comment above about how `_constructors` now has a full definition instead of a tagName